src/main/scala/mesosphere/marathon/state/Group.scala

```diff
@@ -96,7 +97,7 @@ case class Group(
       // replace it. Otherwise do not replace it. Validation will catch conflicting app/group IDs later.
       groups = groups.filter { group => group.id != appDef.id || group.containsApps },
       // replace potentially existing app definition
-      apps = apps.filter(_.id != appDef.id) + appDef
+      apps = apps - appDef.id + ((appDef.id, appDef))
```

Comment:
nit: I think that `appDef.id -> appDef` makes it clearer that you're adding a tuple to the map.